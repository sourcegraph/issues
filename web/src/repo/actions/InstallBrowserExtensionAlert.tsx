import React from 'react'
import CloseIcon from 'mdi-react/CloseIcon'
import ExportIcon from 'mdi-react/ExportIcon'
import * as GQL from '../../../../shared/src/graphql/schema'
import { serviceTypeDisplayNameAndIcon } from './GoToCodeHostAction'

interface Props {
    onAlertDismissed: () => void
    externalURLs: GQL.IExternalLink[]
}

export const InstallBrowserExtensionAlert: React.FunctionComponent<Props> = ({ onAlertDismissed, externalURLs }) => {
    const { displayName, icon } = serviceTypeDisplayNameAndIcon(externalURLs[0]?.serviceType)

    const Icon = icon || ExportIcon

    return (
        <div className="alert alert-info m-2 d-flex justify-content-between">
            <div className="d-flex align-items-center">
                <div className="position-relative">
                    <div className="install-browser-extension-alert__icon-flash" />
                    <Icon className="install-browser-extension-alert__icon" />
                </div>
                <p className="install-browser-extension-alert__text my-0 mr-3">
                    <a href="/help/integration/browser_extension" className="alert-link">
                        Install the Sourcegraph browser extension
                    </a>{' '}
                    to add code intelligence to PRs and file views in {displayName} or any other connected code host.
                </p>
            </div>
            <button type="button" onClick={onAlertDismissed} aria-label="Close alert" className="btn btn-icon">
                <CloseIcon className="icon-inline" />
            </button>
        </div>
    )
}
