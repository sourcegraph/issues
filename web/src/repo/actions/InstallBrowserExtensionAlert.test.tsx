import React from 'react'
import { InstallBrowserExtensionAlert } from './InstallBrowserExtensionAlert'
import { mount } from 'enzyme'
import { noop } from 'lodash'

describe('InstallBrowserExtensionAlert', () => {
    test('GitHub', () => {
        expect(
            mount(
                <InstallBrowserExtensionAlert
                    onAlertDismissed={noop}
                    externalURLs={[
                        {
                            __typename: 'ExternalLink',
                            url: 'https://github.com/sourcegraph/sourcegraph',
                            serviceType: 'github',
                        },
                    ]}
                />
            )
        ).toMatchSnapshot()
    })

    test('GitLab', () => {
        expect(
            mount(
                <InstallBrowserExtensionAlert
                    onAlertDismissed={noop}
                    externalURLs={[
                        {
                            __typename: 'ExternalLink',
                            url: 'https://gitlab.com/rluna-open-source/code-management/sourcegraph/sourcegraph',
                            serviceType: 'gitlab',
                        },
                    ]}
                />
            )
        ).toMatchSnapshot()
    })
})
