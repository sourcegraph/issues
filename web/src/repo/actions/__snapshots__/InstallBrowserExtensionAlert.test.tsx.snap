// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InstallBrowserExtensionAlert GitHub 1`] = `
<InstallBrowserExtensionAlert
  externalURLs={
    Array [
      Object {
        "__typename": "ExternalLink",
        "serviceType": "github",
        "url": "https://github.com/sourcegraph/sourcegraph",
      },
    ]
  }
  onAlertDismissed={[Function]}
>
  <div
    className="alert alert-info m-2 d-flex justify-content-between install-browser-extension-alert"
  >
    <div
      className="d-flex align-items-center"
    >
      <div
        className="position-relative"
      >
        <div
          className="install-browser-extension-alert__icon-flash"
        />
        <Memo(GithubIcon)
          className="install-browser-extension-alert__icon"
        />
      </div>
      <p
        className="install-browser-extension-alert__text my-0 mr-3"
      >
        Get code intelligence while browsing files and reading
         
        PR
        s on 
        GitHub
         or any other connected code host.
         
        <a
          className="alert-link"
          href="/help/integration/browser_extension"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more about Sourcegraph Chrome and Firefox extensions
        </a>
      </p>
    </div>
    <button
      aria-label="Close alert"
      className="btn btn-icon test-close-alert"
      onClick={[Function]}
      type="button"
    >
      <Memo(CloseIcon)
        className="icon-inline"
      />
    </button>
  </div>
</InstallBrowserExtensionAlert>
`;

exports[`InstallBrowserExtensionAlert GitLab 1`] = `
<InstallBrowserExtensionAlert
  externalURLs={
    Array [
      Object {
        "__typename": "ExternalLink",
        "serviceType": "gitlab",
        "url": "https://gitlab.com/rluna-open-source/code-management/sourcegraph/sourcegraph",
      },
    ]
  }
  onAlertDismissed={[Function]}
>
  <div
    className="alert alert-info m-2 d-flex justify-content-between install-browser-extension-alert"
  >
    <div
      className="d-flex align-items-center"
    >
      <div
        className="position-relative"
      >
        <div
          className="install-browser-extension-alert__icon-flash"
        />
        <Memo(GitlabIcon)
          className="install-browser-extension-alert__icon"
        />
      </div>
      <p
        className="install-browser-extension-alert__text my-0 mr-3"
      >
        Get code intelligence while browsing files and reading
         
        MR
        s on 
        GitLab
         or any other connected code host.
         
        <a
          className="alert-link"
          href="/help/integration/browser_extension"
          rel="noopener noreferrer"
          target="_blank"
        >
          Learn more about Sourcegraph Chrome and Firefox extensions
        </a>
      </p>
    </div>
    <button
      aria-label="Close alert"
      className="btn btn-icon test-close-alert"
      onClick={[Function]}
      type="button"
    >
      <Memo(CloseIcon)
        className="icon-inline"
      />
    </button>
  </div>
</InstallBrowserExtensionAlert>
`;
